providers/saml.go

```diff
@@ -135,9 +137,9 @@ func (s *SAMLProvider) initialiseSAMLMiddleware() {
 		SAMLLogger.Debugf("SP acs URL: %v", acsURL.String())
 
 		var forceAuthn = s.config.ForceAuthentication
-
 		sp := saml.ServiceProvider{
-			EntityID:          metadataURL.String(),
+			// if s.config.EntityId is empty, it will default to metadataUrl
+			EntityID:          s.config.EntityId,
```

Comment:
Where is thee code which sets default value to `metadataURL.String()`?

Reply:
@buger inside the library, there we can find:

```
type ServiceProvider struct {
	// Entity ID is optional - if not specified then MetadataURL will be used
	EntityID string

	.
        .
        .
}
```

Reply:
So this line we had previously did nothing?

Reply:
Yes, it wasn't needed.